integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/JDBCRelationVisitor.java

```diff
@@ -0,0 +1,108 @@
+  }
+
+  @Override
+  public boolean isDefinedAt(LogicalPlan x) {
+    return jdbcRelationSupplier(x).isPresent();
+  }
+
+  private Optional<Supplier<JDBCRelation>> jdbcRelationSupplier(LogicalPlan plan) {
+    if (plan instanceof SaveIntoDataSourceCommand) {
+      SaveIntoDataSourceCommand command = (SaveIntoDataSourceCommand) plan;
+      if (command.dataSource() instanceof JdbcRelationProvider) {
+        return Optional.of(
+            () ->
+                (JDBCRelation)
+                    ((JdbcRelationProvider) (command).dataSource())
+                        .createRelation(sqlContext, command.options()));
+      }
+    } else if (plan instanceof LogicalRelation
+        && ((LogicalRelation) plan).relation() instanceof JDBCRelation) {
+      return Optional.of(() -> (JDBCRelation) ((LogicalRelation) plan).relation());
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public List<Dataset> apply(LogicalPlan x) {
+    return jdbcRelationSupplier(x)
+        .map(
+            s -> {
+              JDBCRelation relation = s.get();
+              // TODO- if a relation is composed of a complex sql query, we should attempt to
+              // extract the
+              // table names so that we can construct a true lineage
+              String tableName =
+                  relation
+                      .jdbcOptions()
+                      .parameters()
+                      .get(JDBCOptions.JDBC_TABLE_NAME())
+                      .getOrElse(
+                          new AbstractFunction0<String>() {
+                            @Override
+                            public String apply() {
+                              return "COMPLEX";
+                            }
+                          });
+              URI connectionUri = URI.create(relation.jdbcOptions().url());
+              String namespace =
+                  String.format("%s://%s", connectionUri.getScheme(), connectionUri.getHost());
+              String name = String.format("%s.%s", connectionUri.getPath(), tableName);
+              return Collections.singletonList(
+                  Dataset.builder()
+                      .namespace(namespace)
+                      .name(name)
+                      .facets(
+                          DatasetFacet.builder()
+                              .schema(PlanUtils.schemaFacet(relation.schema()))
+                              .dataSource(
+                                  DatasourceDatasetFacet.builder()
+                                      ._producer(
+                                          URI.create(
+                                              "https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client"))
+                                      ._schemaURL(
+                                          URI.create(
+                                              "https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/spec/OpenLineage.yml#DatasourceDatasetFacet"))
```

Comment:
same comment and let's factor this logic out

Reply:
Sure, I copied these URLs from elsewhere in the code. I'll refactor all the references to a utility class.